[air/docs] Update Trainer documentation

richardliaw opened the pull request:
## Why are these changes needed?
This is stacked on https://github.com/ray-project/ray/pull/27389/files.

This PR adds the following sections:

* a new landing page.
* key concepts. There's quite a bit of redundancy here with the AIR docs. Maybe that's ok for now?
* getting started. 

The addition of these sections bring the Train section in line with the rest of the Ray ecosystem libraries.


## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

ericl reviewed (changes requested):
I made an editing pass over the intro, key concepts, and guides. I think it's OK.

The main thing that seems off right now is the "API Reference" section-- is it just redundant with the AIR API reference? Maybe we should just clarify that and just have two sections:
- Link to AIR API reference
- Deprecated APIs

dmatrix commented on doc/source/train/dl_guide.rst:
```diff
@@ -1,20 +1,9 @@
+This guide explains how to use Train to scale PyTorch, TensorFlow and Horovod.
 
 In this guide, we cover examples for the following use cases:
 
```
How do I ...  to use Ray Train?

richardliaw commented:
@ericl I deduped and addressed comments. Though to me it still feels more natural to just move all the trainer stuff into Ray Train instead of leaving it hanging in Ray AIR package ref.
